samples/drivers/counter/alarm/src/main.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+#include <device.h>
+#include <counter.h>
+#include <misc/printk.h>
+
+#define DELAY 2000000
+#define ALARM_CHANNEL_ID 0
+
+struct counter_alarm_cfg alarm_cfg;
+
+static void test_counter_interrupt_fn(struct device *counter_dev,
+				u8_t chan_id, u32_t ticks, void *user_data)
+{
+	u32_t now_ticks = counter_read(counter_dev);
+	u64_t now_usec = counter_ticks_to_us(counter_dev, now_ticks);
+	int now_sec = (int)(now_usec / USEC_PER_SEC);
+	struct counter_alarm_cfg *config = user_data;
+
+	printk("!!! Alarm !!!\n");
+	printk("Now: %d\n", now_sec);
+
+	counter_disable_channel_alarm(counter_dev, ALARM_CHANNEL_ID);
+
+	config->ticks = config->ticks +	counter_us_to_ticks(counter_dev, DELAY);
```

Comment:
ticks are relative as set in line 47 thus you don't need to do that. If you want to keep precise timing you should set `absolute` to true and keep that line.

Reply:
This was actually the purpose of the sample, cf README:
> At each alarm expiry, a new alarm is configured with a delay increased by 2 seconds.

Of course, we can remove an set always the same duration, but I wanted to play with alarm duration. Let me double the delay at each expiry to make it easier to get.

Reply:
ok, i see. i missed that.